src/app-layer-ftp.c

```diff
@@ -425,6 +453,8 @@ static int FTPParseRequest(Flow *f, void *ftp_state,
                 memcpy(state->port_line, state->current_line,
                         state->current_line_len);
                 state->port_line_len = state->current_line_len;
+                // TODO need to handle ipv6
+                FTPParseGetDynPort(state, input, input_len);
```

Comment:
Because I hope the method FTPParseGetDynPort can be used in FTP active and passive both.  And sorry, i don't know rust, so i don't know how to handle it.